packages/yew/src/suspense/hooks.rs

```diff
@@ -0,0 +1,36 @@
+#[cfg_attr(documenting, doc(cfg(any(target_arch = "wasm32", feature = "tokio"))))]
+#[cfg(any(target_arch = "wasm32", feature = "tokio"))]
+mod feat_futures {
+    use std::future::Future;
+
+    use yew::prelude::*;
+    use yew::suspense::{Suspension, SuspensionResult};
+
+    /// This hook is used to await a future in a suspending context.
+    ///
+    /// A [Suspension] is created from the passed future and the result of the future
+    /// is the output of the suspension.
+    #[hook]
+    pub fn use_future<T, F>(f: F) -> SuspensionResult<T>
```

Comment:
This should probably take a `FnOnce() -> F` to clarify that the future is not always constructed, and only needed on the first render (like `use_state`). See also the comment by futursolo, I also think a `use_future_with_deps` would be a good addition (in a future PR) where this'd be a `FnOnce(&D) -> F` or similar.

You could return something like `SuspensionResult<UseFutureHandle<T>>`, mimicking `use_state`. This would avoid the clone and allow deref to access a `&T`.

Extended comment about `use_future_with_deps`: the hard part is getting the sequencing correct. If dependencies expire quickly, this would spawn multiple futures and "older" futures in that sequence should not override the result of the newest spawned future. No need for that here, since only one future is spawned, ever.

Reply:
It takes a closure now.

Re use_future_with deps: See https://github.com/yewstack/yew/pull/2609#issuecomment-1099429267